### ISSUE TITLE
Where clause points to base class

### ISSUE THREAD

**Issue by adisos** [appfolio/store_base_sti_class#19]
Given the following class definitions,

```ruby
class Address
  belongs_to :addressable, :polymorphic => true
end

class Person
  has_many :addresses, :as => addressable
end

class Vendor < Person
end
```

and given the following code,

```ruby
vendor = Vendor.create(...)
address = vendor.addresses.create(...)

p Address.where(addressable: address).to_sql
```

will output,

```sql
SELECT `addresses`.* FROM `addresses` WHERE `addresses`.`addressable` = 'Person' AND `addresses`.`addressable_id` = 1
```

**Comment by n-rodriguez** [comment]
See https://github.com/appfolio/store_base_sti_class/pull/21

**Comment by jonkessler** [comment]
Fixed by #22
